packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -0,0 +1,243 @@
+﻿import type {
+  Informer,
+  KubernetesObject,
+  ObjectCache,
+  V1Deployment,
+  V1Pod,
+  V1ReplicaSet,
+} from '@kubernetes/client-node';
+import { AppsV1Api, CoreV1Api, KubeConfig, makeInformer } from '@kubernetes/client-node';
+import type { KubeContext } from './kubernetes-context.js';
+import type { ConfigurationRegistry } from './configuration-registry.js';
+import type { ApiSenderType } from './api.js';
+
+interface ContextInternalState {
+  podInformer?: Informer<V1Pod> & ObjectCache<V1Pod>;
+  deploymentInformer?: Informer<V1Deployment> & ObjectCache<V1Deployment>;
+  replicasetInformer?: Informer<V1ReplicaSet> & ObjectCache<V1ReplicaSet>;
+}
+
+export interface ContextState {
+  reachable: boolean;
+  podsCount: number;
+  deploymentsCount: number;
+  replicasetsCount: number;
+}
+
+export class ContextsState {
+  private kubeConfig = new KubeConfig();
+  private contextsState = new Map<string, ContextState>();
+  private contextsInternalState = new Map<string, ContextInternalState>();
+
+  constructor(
+    private readonly configurationRegistry: ConfigurationRegistry,
+    private readonly apiSender: ApiSenderType,
+  ) {}
+
+  async update(enabled: boolean | undefined, kubeconfig: KubeConfig) {
+    this.kubeConfig = kubeconfig;
+    if (enabled) {
+      // Add informers for new contexts
+      for (const context of this.kubeConfig.contexts) {
+        if (this.contextsInternalState.get(context.name) === undefined) {
+          this.contextsState.set(context.name, {
+            reachable: false,
+            podsCount: 0,
+            deploymentsCount: 0,
+            replicasetsCount: 0,
+          });
+          const informers = this.createKubeContextInformers(context);
+          if (informers) {
+            this.contextsInternalState.set(context.name, informers);
+          }
+        }
+      }
+      // Delete informers for removed contexts
+      for (const [name, state] of this.contextsInternalState) {
+        if (!this.kubeConfig.contexts.find(c => c.name === name)) {
+          await state.podInformer?.stop();
+          await state.deploymentInformer?.stop();
+          await state.replicasetInformer?.stop();
+          this.contextsInternalState.delete(name);
+          this.contextsState.delete(name);
+        }
+      }
+    } else {
+      for (const state of this.contextsInternalState.values()) {
+        await state.podInformer?.stop();
+        await state.deploymentInformer?.stop();
+        await state.replicasetInformer?.stop();
+      }
+      this.contextsInternalState.clear();
+      this.contextsState.clear();
+      this.dispatchContextsState();
+    }
+    console.log('==> # informers', this.contextsInternalState.size, this.contextsState.size);
```

Comment:
Replace `==>` with meaningful, human readable message.

Reply:
I'm using this `==>` sign to grep into it, to specifically get my logs during debug sessions. Most of these logs will probably disappear before the final revision